packages/neuron-wallet/src/services/settings.ts

```diff
@@ -78,11 +87,15 @@ export default class SettingsService extends Store {
       JSON.stringify({
         locale: app.getLocale(),
         ckbDataPath: path.resolve(app.getPath('userData'), 'chains/mainnet'),
+        isFirstSync: true,
       })
     )
     if (!this.getNodeDataPath(LIGHT_CLIENT_MAINNET) || !this.getNodeDataPath('ckb')) {
       this.migrateDataPath()
     }
+    if (this.isFirstSync === undefined) {
+      this.isFirstSync = !fs.existsSync(path.join(this.getNodeDataPath(), 'ckb.toml'))
+    }
```

Comment:
If the users have installed and run Neuron before, the config will not initialize again. If the `isFirstSync ` is true, the users need to click start sync, then the config will initialize then the `isFirstSync` will be set to `false`.

Reply:
I got your point, but `isFirstSync` could be a runtime variable

When Neuron is re-installed, the initialization of ckb node configuration won't be performed, so the `isFirstSync` will be `false` and no information will be displayed.

Reply:
Re-installing the Neuron will not change the config file, If the user deletes all of the Neuron caches, the `isFirstSync ` will be true and the ckb data path will be empty.

Reply:
> Re-installing the Neuron will not change the config file, If the user deletes all of the Neuron caches, the `isFirstSync ` will be true and the ckb data path will be empty.

Will `delete all of the neuron caches` remove the `ckb.toml` file? If not, the `isFirstSync` is still `false` because initialization won't be run again.

Reply:
With users manually operating, there may exist four conditions:
1. `isFirstSync ` is true, and `ckb. toml` exists or does not exist. I guess notify the user there is no problem.
2. `isFirstSync ` is false, and `ckb. toml` exists or does not exist. Neuron must have notified the user before, so we do not notify the user is normal. The neuron will init the ckb if the `ckb.toml` does not exist.

Reply:
> With users manually operating, there may exist four conditions:
> 
> 1. `isFirstSync ` is true, and `ckb. toml` exists or does not exist. I guess notify the user there is no problem.
> 2. `isFirstSync ` is false, and `ckb. toml` exists or does not exist. Neuron must have notified the user before, so we do not notify the user is normal. The neuron will init the ckb if the `ckb.toml` does not exist.

In what case will `isFirstSync = true` and `ckb.toml` exists?

---

IMO, the `isFirstSync` could be conducted from the existence of `ckb.toml`, it can be a runtime variable instead of a data read from disk.

Each time Neuron tries to sync with the full node, full node tries to generate the configuration file.

If the configuration file is created successfully, it's the first time to sync because synchronization cannot happen without a configuration file.

If the configuration file is not created successfully, either it's been generated before, or ckb fails to run.

If the configuration has been generated before, the notification should have been displayed.

Reply:
I have tried setting the `isFirstSync` when the ckb init. But it will break the original ckb runner logic. And every time the user resets the ckb data path, it needs the user's confirmation.